src/models/Factor.js

```diff
@@ -244,20 +244,21 @@ function (Okta, Q, factorUtil, BaseLoginModel) {
             'delay': PUSH_INTERVAL
           };
           if (trans.status === 'MFA_CHALLENGE' && trans.poll) {
+            // In verify case since we poll we dont return the trans
+            // rather set it and continue to poll
+            setTransaction(trans);
             return Q.delay(PUSH_INTERVAL).then(function() {
               if (self.pushFactorHasAutoPush()) {
                 options.autoPush = function() {
                   return self.get('autoPush');
                 };
               }
-              return trans.poll(options);
+              return trans.poll(options).then(function(trans) {
+                setTransaction(trans);
+              });
             });
           }
-          return trans;
-        })
-        .fail(function (err) {
-          // Clean up the cookie on failure.
-          throw err;
+          setTransaction(trans);
```

Comment:
Can you move the setTransaction to before the if? The logic should be:
- We always set the current transaction (first)
- If it's a poll, we then start the poll and update the transaction when it's done